src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -121,20 +121,33 @@ public void removeCustomer(Customer key) {
      */
     public boolean hasCommission(Customer customer, Commission commission) {
         requireAllNonNull(customer, commission);
+        // Previously duplicate commission won't be detected because customer field has not been set.
+        commission.setCustomer(customer);
         return getCustomerCommissions(customer).contains(commission);
     }
 
     /**
      * Adds a commission to the address book.
      * The commission must not already exist in the customer's commission list.
      */
-    public void addCommission(Customer customer, Commission commission) {
+    public Customer addCommission(Customer customer, Commission commission) {
         requireAllNonNull(customer, commission);
+        // Hashset would cause the commissions to be out of order
         HashSet<Commission> customerCommissions = new HashSet<>(getCustomerCommissions(customer));
         customerCommissions.add(commission);
+        commission.setCustomer(customer);
         commissions.add(commission);
+
         Customer newCustomer = customer.copyWithCommissions(customerCommissions);
+        // Not efficient but ensures all commissions have reference to the new customer who replaced the old one
+        // More ideal to keep same customer reference and update the object instead @Amar suggestion to add counter
```

Comment:
I agree with this point: we shouldn't need to constantly update the entire list of commissions just to trigger the redraw of the UI.

Reply:
kinda random (also haven't tested or tried using it and probably not necessary yet in this PR) but apparently `ListView` (the thing `CustomerListPanel` has that uses the `observableList` has this interesting method)

## `ListView.refresh()`

```
 * Calling {@code refresh()} forces the ListView control to recreate and
 * repopulate the cells necessary to populate the visual bounds of the control.
 * In other words, this forces the ListView to update what it is showing to
 * the user. This is useful in cases where the underlying data source has
 * changed in a way that is not observed by the ListView itself.
```